nvi-rest/src/test/java/no/sikt/nva/nvi/fetch/FetchNviCandidateHandlerTest.java

```diff
@@ -0,0 +1,127 @@
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import com.amazonaws.services.lambda.runtime.Context;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import no.sikt.nva.nvi.common.model.business.ApprovalStatus;
+import no.sikt.nva.nvi.common.model.business.Candidate;
+import no.sikt.nva.nvi.common.model.business.Creator;
+import no.sikt.nva.nvi.common.model.business.Level;
+import no.sikt.nva.nvi.common.model.business.Note;
+import no.sikt.nva.nvi.common.model.business.PublicationDate;
+import no.sikt.nva.nvi.common.model.business.Status;
+import no.sikt.nva.nvi.common.model.business.Username;
+import no.sikt.nva.nvi.common.service.NviService;
+import no.unit.nva.testutils.HandlerRequestBuilder;
+import nva.commons.apigateway.GatewayResponse;
+import nva.commons.core.Environment;
+import org.apache.hc.core5.http.ContentType;
+import org.apache.hc.core5.http.HttpStatus;
+import org.eclipse.jetty.http.HttpHeader;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class FetchNviCandidateHandlerTest {
+
+    private static final Context CONTEXT = mock(Context.class);
+    private ByteArrayOutputStream output;
+    private FetchNviCandidateHandler handler;
+    private Environment environment;
+    private NviService service;
+
+    @BeforeEach
+    public void setUp() {
+        environment = mock(Environment.class);
+        when(environment.readEnv(ALLOWED_ORIGIN_ENV)).thenReturn("*");
+
+        output = new ByteArrayOutputStream();
+        service = mock(NviService.class);
+        handler = new FetchNviCandidateHandler(service, environment);
+    }
+
+    @Test
+    void shouldReturn400whenMissingCandidateId() throws IOException {
+
+        var input = new HandlerRequestBuilder<InputStream>(dtoObjectMapper)
+                        .withHeaders(Map.of(HttpHeader.ACCEPT.asString(), ContentType.APPLICATION_JSON.getMimeType()))
+                        .build();
+        handler.handleRequest(input, output, CONTEXT);
+        var gatewayResponse = getGatewayResponse();
+
+        assertEquals(HttpStatus.SC_NOT_FOUND, gatewayResponse.getStatusCode());
+    }
+
+    @Test
+    void shouldReturn404whenNoCandidateFound() throws IOException {
+
+        var candidateIdentifier = randomUri();
+        InputStream input = getInput(candidateIdentifier);
+        when(service.getCandidate(URI.create(""))).thenReturn(Optional.empty());
+        handler.handleRequest(input, output, CONTEXT);
+        var gatewayResponse = getGatewayResponse();
+
+        assertEquals(HttpStatus.SC_NOT_FOUND, gatewayResponse.getStatusCode());
+    }
+
+    @Test
+    void shouldReturnValidCandidateIfExists() throws IOException {
```

Comment:
Two things here:

- shouldReturnCandidateWhenExistingCandidateIsRequested
- valid…you check one thing…should we be validating candidates on output or input? We probably need to verify the structure of candidates in some test…we have done this previously with does-not-contain-empty-or-null-values-or-whatever-it-is-called